From afa3aa9f60c499a7cbbd0ad63ad460c9d11712ec Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:32:55 -0400
Subject: [infiniband] qib: Add missing serdes init sequence

Message-id: <e786bff6ed02b715293b3f371882bfbd8645f8ac.1411579814.git.dledford@redhat.com>
Patchwork-id: 96703
O-Subject: [Patch RHEL7 02/19] IB/qib: Add missing serdes init sequence
Bugzilla: 1085961
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Mike Marciniszyn <mike.marciniszyn@intel.com>

Research has shown that commit a77fcf895046 ("IB/qib: Use a single
txselect module parameter for serdes tuning") missed a key serdes init
sequence.

This patch add that sequence.

Cc: <stable@vger.kernel.org>
Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 2f75e12c4457a9b3d042c0a0d748fa198dc2ffaf)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/qib/qib_iba7322.c b/drivers/infiniband/hw/qib/qib_iba7322.c
index 5bfc02f..d1bd213 100644
--- a/drivers/infiniband/hw/qib/qib_iba7322.c
+++ b/drivers/infiniband/hw/qib/qib_iba7322.c
@@ -2395,6 +2395,11 @@ static int qib_7322_bringup_serdes(struct qib_pportdata *ppd)
  qib_write_kreg_port(ppd, krp_ibcctrl_a, ppd->cpspec->ibcctrl_a);
  qib_write_kreg(dd, kr_scratch, 0ULL);
 
+ /* ensure previous Tx parameters are not still forced */
+ qib_write_kreg_port(ppd, krp_tx_deemph_override,
+  SYM_MASK(IBSD_TX_DEEMPHASIS_OVERRIDE_0,
+  reset_tx_deemphasis_override));
+
  if (qib_compat_ddr_negotiate) {
   ppd->cpspec->ibdeltainprog = 1;
   ppd->cpspec->ibsymsnap = read_7322_creg32_port(ppd,
-- 
1.7.1